### PR TITLE
Added futher docstrings and typehints to :class:`~.Mobject`

### PR DESCRIPTION
**Pull request by kolibril13**
## Checklist
- [ ] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [ ] I have written a descriptive PR title (see top of PR template for examples)
- [ ] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -1749,9 +1753,11 @@ def get_critical_point(self, direction):
     # Pseudonyms for more general get_critical_point method
 
     def get_edge_center(self, direction) -> np.ndarray:
+        """Get edge coordinates for certain direction"""
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1762]:
```suggestion
        """Get edge coordinates for certain direction."""
```

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -1749,9 +1753,11 @@ def get_critical_point(self, direction):
...
         return self.get_critical_point(direction)
 
     def get_corner(self, direction) -> np.ndarray:
+        """Get corner coordinates for certain direction"""
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1766]: same

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -1782,10 +1788,12 @@ def get_left(self) -> np.ndarray:
         """Get left coordinates of a box bounding the :class:`~.Mobject`"""
         return self.get_edge_center(LEFT)
 
-    def get_zenith(self):
+    def get_zenith(self) -> np.ndarray:
+        """Get zenith coordinates of a box bounding a 3D :class:`~.Mobject`"""
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1798]: same `.`

File: manim/mobject/mobject.py [hunk 4]
```diff
@@ -1898,7 +1913,8 @@ def align_to(
         direction=ORIGIN,
         alignment_vect=UP,
     ):
-        """Examples:
+        """Aligns mobject to another :class:`~.Mobject` in a certain direction.
+        Examples:
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1923]: There needs to be a new line between these two lines

File: manim/mobject/mobject.py [hunk 5]
```diff
@@ -58,6 +58,10 @@ class Mobject(Container):
     points : :class:`numpy.ndarray`
         The points of the objects.
 
+        .. seealso::
+
+        :class:`VMobject`
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:63]:
```suggestion
        :class:`~.VMobject`
```

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]
Minor changes needed. Remember punctuation/ `~.`, there's a few other places I didn't add the punctuation.

File: manim/mobject/mobject.py [hunk 6]
```diff
@@ -1755,9 +1759,11 @@ def get_critical_point(self, direction):
     # Pseudonyms for more general get_critical_point method
 
     def get_edge_center(self, direction) -> np.ndarray:
+        """Get edge coordinates for certain direction"""
         return self.get_critical_point(direction)
 
     def get_corner(self, direction) -> np.ndarray:
+        """Get corner coordinates for certain direction"""
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1766]:
```suggestion
        """Get corner coordinates for certain direction."""
```

File: manim/mobject/mobject.py [hunk 7]
```diff
@@ -1788,10 +1794,12 @@ def get_left(self) -> np.ndarray:
         """Get left coordinates of a box bounding the :class:`~.Mobject`"""
         return self.get_edge_center(LEFT)
 
-    def get_zenith(self):
+    def get_zenith(self) -> np.ndarray:
+        """Get zenith coordinates of a box bounding a 3D :class:`~.Mobject`"""
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1798]:
```suggestion
        """Get zenith coordinates of a box bounding a 3D :class:`~.Mobject`."""
```

File: manim/mobject/mobject.py [hunk 8]
```diff
@@ -1904,7 +1919,8 @@ def align_to(
         direction=ORIGIN,
         alignment_vect=UP,
     ):
-        """Examples:
+        """Aligns mobject to another :class:`~.Mobject` in a certain direction.
+        Examples:
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1923]:
```suggestion
        """Aligns mobject to another :class:`~.Mobject` in a certain direction.
        
        Examples:
```

File: manim/mobject/mobject.py [hunk 9]
```diff
@@ -1788,10 +1794,12 @@ def get_left(self) -> np.ndarray:
...
         return self.get_edge_center(LEFT)
 
-    def get_zenith(self):
+    def get_zenith(self) -> np.ndarray:
+        """Get zenith coordinates of a box bounding a 3D :class:`~.Mobject`."""
         return self.get_edge_center(OUT)
 
-    def get_nadir(self):
+    def get_nadir(self) -> np.ndarray:
+        """Get nadir (opposite the zenith) coordinates of a box bounding a 3D :class:`~.Mobject`"""
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1802]:
```suggestion
        """Get nadir (opposite the zenith) coordinates of a box bounding a 3D :class:`~.Mobject`."""
```

**Review by jsonvillanueva — APPROVED** [review]
LGTM

**Pull request merged**